pkg/cmd/secrets.go

```diff
@@ -538,8 +538,29 @@ func substituteSecrets(cmd *cobra.Command, args []string) {
 		}
 	}
 
+	// parse template file
+	templateBody := controllers.ReadTemplateFile(args[0])
+	template, e := controllers.ParseSecretsTemplate(templateBody)
+	if !e.IsNil() {
+		utils.HandleError(e.Unwrap(), e.Message)
+	}
+
+	fetchAllSecrets := utils.GetBoolFlag(cmd, "fetch-all")
+	var secretNames []string
+	if !fetchAllSecrets {
+		// parse secrets from template
+		for _, field := range controllers.TemplateFields(template) {
+			if strings.HasPrefix(field, "{{.") && strings.HasSuffix(field, "}}") {
+				f := strings.TrimPrefix(field, "{{.")
+				f = strings.TrimSuffix(f, "}}")
+				secretNames = append(secretNames, f)
+			}
+		}
+	}
```

Comment:
Unfortunately go doesn't provide a better way of grabbing the parsed fields. Thus we provide the `fetch-all` flag (open to alternative naming) to override our new parsing logic and instead fetch all secrets. This allows us to continue to support more advanced template logic than what we can manually parse.

Reply:
I'm worried about changing the default to this new behavior because it's a breaking change. I know we have a least a few users who write dynamic templates that don't reference secrets directly at all. For example,

```
{{ range $key, $value := . }}
{{ $key }}==={{ $value }}
{{ end }}
```

Reply:
Really glad you have context on usage in the real world. There's still an option to parse the template fields and, if all fields are of the format "{{.SECRET_NAME}}", then fetch those parsed secrets. I'm not sure this complexity is worth it though, and would be fine abandoning this feature. Wdyt?

Reply:
Agreed that parsing the template ourselves isn't viable. I'm going to hold off on supporting this in `doppler secrets substitute` for now.